core/src/avm1/scope.rs

```diff
@@ -60,67 +59,31 @@ impl<'gc> Scope<'gc> {
     /// Construct a scope for use with `tellTarget` code where the timeline
     /// scope has been replaced with another given object.
     pub fn new_target_scope(
-        mut parent: GcCell<'gc, Self>,
+        parent: Gc<'gc, Self>,
         clip: Object<'gc>,
         mc: MutationContext<'gc, '_>,
-    ) -> GcCell<'gc, Self> {
-        let mut bottom_scope = None;
-        let mut top_scope: Option<GcCell<'gc, Self>> = None;
+    ) -> Gc<'gc, Self> {
```

Comment:
`new_target_scope` can actually return `Self`, then the `mc` parameter is redundant.

Reply:
`new_target_scope` needs the `mc` parameters to allocate the parent scopes. Will change for `new_with_scope` though.